src/core/server/http/http_server.ts

```diff
@@ -25,21 +25,27 @@ import { createServer, getListenerOptions, getServerOptions } from './http_tools
 import { adoptToHapiAuthFormat, AuthenticationHandler } from './lifecycle/auth';
 import { adoptToHapiOnPostAuthFormat, OnPostAuthHandler } from './lifecycle/on_post_auth';
 import { adoptToHapiOnPreAuthFormat, OnPreAuthHandler } from './lifecycle/on_pre_auth';
-import { Router, KibanaRequest, ResponseHeaders } from './router';
+import { KibanaRequest, LegacyRequest, ResponseHeaders, Router } from './router';
 import {
   SessionStorageCookieOptions,
   createCookieSessionStorageFactory,
 } from './cookie_session_storage';
 import { SessionStorageFactory } from './session_storage';
-import { AuthStateStorage } from './auth_state_storage';
-import { AuthHeadersStorage } from './auth_headers_storage';
+import { AuthStateStorage, GetAuthState, IsAuthenticated } from './auth_state_storage';
+import { AuthHeadersStorage, GetAuthHeaders } from './auth_headers_storage';
 import { BasePath } from './base_path_service';
 
+/** @public */
```

Comment:
I think that either this type or the `HttpServiceSetup` type should have a high-level explanation of the capabilities of the HttpService, why you would use it, and a couple examples of how to use the different APIs together.

The examples in each individual function are useful, but they don't help someone who is learning about this service as a whole for the first time because they don't know which methods to even look at. I actually think the docs you added to MIGRATION.md would be great here. You could link to the generated markdown from the migration guide if you want.